pkg/spdk/replica.go

```diff
@@ -1043,6 +1043,110 @@ func (r *Replica) Get() (pReplica *spdkrpc.Replica) {
 	return ServiceReplicaToProtoReplica(r)
 }
 
+func (r *Replica) Expand(spdkClient *spdkclient.Client, size uint64) error {
+	r.Lock()
+	defer r.Unlock()
+
+	r.log.Info("Expanding replica")
+
+	clusterSize, err := r.fetchClusterSize(spdkClient)
+	if err != nil {
+		return err
+	}
+
+	roundedSize := util.RoundUp(size, clusterSize)
+	if roundedSize != size {
+		return fmt.Errorf("replica %s rounded up spec size from %v to %v since the spec size should be multiple of MiB", r.Name, size, roundedSize)
+	}
+
+	if r.SpecSize > roundedSize {
+		return fmt.Errorf("cannot expand replica %s to a smaller size %v, current spec size %v", r.Name, size, r.SpecSize)
+	} else if r.SpecSize == roundedSize {
+		r.log.Infof("Replica %s had been expanded to size %v", r.Name, size)
+		return nil
+	}
+
+	// double check if size is already be expanded
+	headBdevLvol, err := spdkClient.BdevLvolGetByName(r.Alias, 0)
+	if err != nil {
+		r.log.Errorf("Get replica %s failed, %v", r.Name, err)
+		return err
+	}
+	bdevLvol := BdevLvolInfoToServiceLvol(&headBdevLvol)
+
+	// resize it if not equal
+	if bdevLvol.SpecSize != size {
```

Comment:
Do we need to check the valid case here `bdevLvol.SpecSize < Size`? for other cases, should not proceed expansion.